x-pack/plugins/security_solution/public/timelines/components/timeline/unified_components/data_table/additional_row.tsx

```diff
@@ -0,0 +1,79 @@
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React, { useRef, memo, useEffect } from 'react';
+
+import type { EuiDataGridSetCellProps, EuiDataGridCellValueElementProps } from '@elastic/eui';
+import { EuiFlexGroup, EuiFlexItem } from '@elastic/eui';
+import type { TimelineItem } from '@kbn/timelines-plugin/common';
+import { EventsTrSupplement } from '../../styles';
+import { StatefulRowRenderer } from '../../body/events/stateful_row_renderer';
+import type { RowRenderer } from '../../../../../../common/types';
+import { useTimelineUnifiedDataTableContext } from './use_timeline_unified_data_table_context';
+
+/** This offset begins at two, because the header row counts as "row 1", and aria-rowindex starts at "1" */
+const ARIA_ROW_INDEX_OFFSET = 2;
+
+type RenderCellValueProps = EuiDataGridCellValueElementProps;
+
+export interface AdditionalRowOwnProps {
+  rowIndex: number;
+  event: TimelineItem;
+  setCellProps?: (props: EuiDataGridSetCellProps) => void;
+  timelineId: string;
+  enabledRowRenderers: RowRenderer[];
+}
+
+export type AdditionalRowProps = RenderCellValueProps & AdditionalRowOwnProps;
+
+const AdditionalRowComp: React.FC<AdditionalRowProps> = memo(
+  ({ rowIndex, event, setCellProps, timelineId, enabledRowRenderers }) => {
+    const containerRef = useRef<HTMLDivElement | null>(null);
+
+    /*
+     * Ideally, unified data table could have handled the styling of trailing columns when a row is expanded.
+     * But, a trailing column can have arbitrary design and that is why it is best for consumer to handle the styling
+     * as we are doing below
+     *
+     * */
+    const ctx = useTimelineUnifiedDataTableContext();
+
+    useEffect(() => {
+      setCellProps?.({
+        className: ctx.expanded?.id === event._id ? 'unifiedDataTable__cell--expanded' : '',
+        style: { width: '100%', height: 'auto' },
+      });
+    }, [ctx.expanded?.id, setCellProps, rowIndex, event._id]);
+
+    if (!enabledRowRenderers || enabledRowRenderers.length === 0) return null;
+
+    return (
+      <EuiFlexGroup
+        justifyContent="center"
+        alignItems="center"
+        data-test-subj={`timeline-row-renderer-${rowIndex}`}
+      >
+        <EuiFlexItem grow={false}>
+          <EventsTrSupplement>
+            <StatefulRowRenderer
+              ariaRowindex={rowIndex + ARIA_ROW_INDEX_OFFSET}
+              containerRef={containerRef}
+              event={event}
+              lastFocusedAriaColindex={rowIndex - 1}
+              rowRenderers={enabledRowRenderers}
+              timelineId={timelineId}
+            />
+          </EventsTrSupplement>
+        </EuiFlexItem>
+      </EuiFlexGroup>
+    );
+  }
+);
+AdditionalRowComp.displayName = 'AdditionalRowComp';
+
+export const AdditionalRow = React.memo(AdditionalRowComp);
```

Comment:
any reason for returning the memo here and have `AdditionalRow` and `AdditionalRowComp`? I feel like having the memo where we defined the constant above and exporting it at the same time is easier. I've never liked having those `XXXComp` or `XXXComponent` constant for no reason...

Reply:
Thanks for catching it. Event I don't like it and i am changing my habits but sometimes forget. Review new changes here : be6bc648b8a